### PR TITLE
ci: add a ML model artifact test case

### PR DESCRIPTION
**Pull request by rchincha**
Both as a test and an example.

Inspired by:
    https://github.com/kubeflow/model-registry/blob/main/docs/logical_model.md

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/blackbox/pushpull_authn.bats [hunk 1]
```diff
@@ -173,3 +173,20 @@ EOF
     [ "$status" -eq 0 ]
     [ $(echo "${lines[-1]}" | jq '.manifests | length') -eq 0 ]
 }
+
+@test "ML artifacts" {
+  # download model data
+  curl -v -L0 https://github.com/tarilabs/demo20231212/raw/main/v1.nb20231206162408/mnist.onnx -o ${BATS_FILE_TMPDIR}/mnist.onnx
+
+  # upload artifact with required annotations and version
+  run regctl artifact put --annotation description="used for demo purposes" --annotation model_format_name="onnx" --annotation model_format_name="onnx" --annotation  model_format_version="1" --artifact-type "application/vnd.model.type" localhost:${zot_port}/models/my-model-from-gh:v1 -f ${BATS_FILE_TMPDIR}/mnist.onnx
```
Comment (andaaron) [inline test/blackbox/pushpull_authn.bats:182]:
```suggestion
  run regctl artifact put --annotation description="used for demo purposes" --annotation model_format_name="onnx" --annotation  model_format_version="1" --artifact-type "application/vnd.model.type" localhost:${zot_port}/models/my-model-from-gh:v1 -f ${BATS_FILE_TMPDIR}/mnist.onnx
```

File: test/blackbox/pushpull_authn.bats [hunk 2]
```diff
@@ -173,3 +173,20 @@ EOF
...
+
+  # list artifacts
+  run regctl artifact list localhost:${zot_port}/models/my-model-from-gh:v1 --format raw-body
+
+  # list artifacts of type
```
Comment (andaaron) [inline test/blackbox/pushpull_authn.bats:187]: Maybe do some checks on the returned outputs of regctl. Make sure it returns data, and it is data we expect.

Comment (rchincha) [reply test/blackbox/pushpull_authn.bats:190]: Fixed.

File: examples/config-minimal.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "distSpecVersion": "1.1.0",
     "storage": {
-        "rootDirectory": "/tmp/zot"
+        "rootDirectory": "/data/rchincha/tmp/zot"
```
Comment (andaaron) [inline examples/config-minimal.json:4]: Revert before merge

**Review by andaaron — COMMENTED** [review]
Earlier you had been using 'run regctl', was the content of 'lines' or 'output' variable not enough to get the shasum or index/manifest? Or why did you switch to using files?

**Comment by rchincha** [comment]
> Earlier you had been using 'run regctl', was the content of 'lines' or 'output' variable not enough to get the shasum or index/manifest? Or why did you switch to using files?

When pulling the artifact, didn't want to hold it in a output var - could be very large.
The rest were removed first to verify output and later to make the usage uniform.

**Review by andaaron — APPROVED** [review]

**Pull request merged**